explorer/files/ncpi-catalog/ncpi-update-utils.ts

```diff
@@ -0,0 +1,42 @@
+import { stringify } from "csv-stringify/sync";
+import fs from "fs";
+import { SOURCE_FIELD_KEY } from "./constants";
+import { dbGapId, NCPIPlatformStudy } from "./entities";
+
+export const sourcePath = "ncpi-catalog/files/dashboard-source-ncpi.tsv";
+
+export function replaceTsv(path: string, data: string[][]): void {
+  const stringifiedOut = stringify(data, { delimiter: "\t" });
+
+  fs.writeFileSync(path, stringifiedOut);
+  console.log("Source updated!");
+}
+
+export function mergeSourceStudies(
+  sourceStudies: NCPIPlatformStudy[],
+  studyType: string,
```

Comment:
Change `studyType` to `platform` and type it as `SOURCE_CATEGORY_KEY` here and elsewhere.